src/utils/validation-server.ts

```diff
@@ -0,0 +1,27 @@
+import { Settings as settings } from '$lib/server/settings';
+import type { ReservationData } from '$types';
+
+import { getOverlappingReservations } from '$utils/validation';
+
```

Comment:
@nrxszvo this should address the backend settings issue

Reply:
I think the call to getOverlappingReservations will fail here because it's still trying to use the client-side Settings store.

I'm starting to wonder if it might be simpler to back out the changes to how Settings is used from this PR and address the issue of how Settings is dealt with on server vs client in a separate PR.  

I also spent some time exploring the idea of a client/server-agnostic getSettings function, and it seemed like it was going to require a very large amount of code updates but might not provide much benefit.

Reply:
that is why i did not proceed in implementing it because i would have to do dynamic imports.. but my thought would be more.

1. we should only validate on submit, then let the server determine what the issue is since it will have the real latest data

Reply:
Ok, let's chat more about it more in person and decide how we want to proceed.  Maybe tomorrow morning?